src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -475,8 +382,8 @@ static void on_resolver_result_changed_locked(void* arg, grpc_error* error) {
       service_config_json =
           gpr_strdup(grpc_channel_arg_get_string(channel_arg));
       if (service_config_json != nullptr) {
-        grpc_service_config* service_config =
-            grpc_service_config_create(service_config_json);
+        grpc_core::UniquePtr<grpc_core::ServiceConfig> service_config =
```

Comment:
nit: calling the constructor avoids temporary rvalues that may be created when using the = syntax.